instrumentation/sinatra/lib/opentelemetry/instrumentation/sinatra/middlewares/tracer_middleware.rb

```diff
@@ -16,17 +16,24 @@ def initialize(app)
           end
 
           def call(env)
-            @app.call(env)
+            response = @app.call(env)
           ensure
-            trace_response(env)
+            trace_response(env, response)
           end
 
-          def trace_response(env)
+          def trace_response(env, response)
             span = OpenTelemetry::Instrumentation::Rack.current_span
             return unless span.recording?
 
             span.set_attribute('http.route', env['sinatra.route'].split.last) if env['sinatra.route']
             span.name = env['sinatra.route'] if env['sinatra.route']
+
+            #response looks like [HTTP code, headers, [response body]]
+            sinatra_response = ::Sinatra::Response.new([], response.first)
```

Comment:
🤔 `response.first`

What does it mean for there to be more than 1 response? Is response an array or enumerable structure?

Reply:
Oh silly me. That's the rack response array.